src/trace/implementations/ord.rs

```diff
@@ -46,67 +49,112 @@ pub type OrdKeySpine<K, T, R, O=usize> = Spine<Rc<OrdKeyBatch<K, T, R, O>>>;
 /// A trace implementation for empty values using a spine of abomonated ordered lists.
 pub type OrdKeySpineAbom<K, T, R, O=usize> = Spine<Rc<Abomonated<OrdKeyBatch<K, T, R, O>, Vec<u8>>>>;
 
+/// A trace implementation backed by columnar storage.
+pub type ColValSpine<K, V, T, R, O=usize> = Spine<Rc<OrdValBatch<K, V, T, R, O, TimelyStack<K>, TimelyStack<V>>>>;
+/// A trace implementation backed by columnar storage.
+pub type ColKeySpine<K, T, R, O=usize> = Spine<Rc<OrdKeyBatch<K,  T, R, O, TimelyStack<K>>>>;
+
+
+/// A container that can retain/discard from some offset onward.
+pub trait RetainFrom<T> {
+    /// Retains elements from an index onwards that satisfy a predicate.
+    fn retain_from<P: FnMut(usize, &T)->bool>(&mut self, index: usize, predicate: P);
+}
+
+impl<T> RetainFrom<T> for Vec<T> {
+    fn retain_from<P: FnMut(usize, &T)->bool>(&mut self, index: usize, mut predicate: P) {
+        let mut write_position = index;
+        for position in index .. self.len() {
+            if predicate(position, &self[position]) {
+                // TODO: compact the inner region and update pointers.
```

Comment:
What does this mean? :D

Reply:
Looks vestigial, as it also appears (with clearer meaning) in the `columnation` code.